src/relay.cpp

```diff
@@ -71,6 +71,153 @@ std::map<int, std::string> counterMap = {{DHCPv6_MESSAGE_TYPE_UNKNOWN, "Unknown"
 /* interface to vlan mapping */
 std::unordered_map<std::string, std::string> vlan_map;
 
+
+/* Class Options */
+
+bool Options::Add(OptionCode key, const uint8_t *value, uint16_t len) {
+    std::vector<uint8_t> option_v(value, value + len);
+    options[key] = option_v;
+    return true;
+}
+
+bool Options::Delete(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return false;
+    }
+    options.erase(key);
+    return true;
+}
+
+std::vector<uint8_t> Options::Get(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return std::vector<uint8_t>{};
+    }
+    return options[key];
+}
+
+std::vector<uint8_t> *Options::MarshalBinary() {
+    if (options.empty()) {
+        return nullptr;
+    }
+    list.clear();
+    for (auto &itr : options) {
+        uint16_t type = htons(itr.first);
+        uint16_t length = htons(itr.second.size());
+        list.push_back(((uint8_t *)&type)[0]);
+        list.push_back(((uint8_t *)&type)[1]);
+        list.push_back(((uint8_t *)&length)[0]);
+        list.push_back(((uint8_t *)&length)[1]);
+        list.insert(list.end(), itr.second.begin(), itr.second.end());
+    }
+    return &list;
+}
+
+bool Options::UnmarshalBinary(const uint8_t *packet, uint16_t length) {
+    while (length >= sizeof(dhcpv6_option)) {
+        auto option = (dhcpv6_option *)packet;
+        auto type = ntohs(option->option_code);
+        if (type > DHCPv6_OPTION_LIMIT) {
+            syslog(LOG_WARNING, "option type %d is invalid \n", type);
+            return false;
+        }
+        auto len = ntohs(option->option_length);
+        if (len + sizeof(dhcpv6_option) > length) {
+            syslog(LOG_WARNING, "unmarshal packet error: option %d length %d over range\n", type, len);
+            return false;
+        }
+        auto value_ptr = packet + sizeof(dhcpv6_option);
+        std::vector<uint8_t> option_v(value_ptr, value_ptr + len);
+        options[type] = option_v;
+        auto offset = sizeof(dhcpv6_option) + len;
+        length -= offset;
+        packet += offset;
+    }
+    return true;
+}
+
+/* Class RelayMsg */
+
+uint8_t *RelayMsg::MarshalBinary(uint16_t &len) {
+    if (buffer == nullptr) {
+        buffer.reset(new (std::nothrow)uint8_t[BUFFER_SIZE]);
+        if (!buffer) {
+            syslog(LOG_ERR, "Failed to init relay msg buffer\n");
+            exit(1);
+        }
+    }
+
+    auto ptr = buffer.get();
+    std::memcpy(ptr, &hdr, sizeof(dhcpv6_relay_msg));
```

Comment:
hdr as a member attribute is not explicit. Can you make it more explicit that this is a member attribute? This comments is applicable to other member attributes in this PR too.

Reply:
Got it, will change member name to use '_' as prefix.
Thanks Ying for comments.

Reply:
What is the common coding style to mark a member attribute in SONIC repos? '_" or 'm_'? Can you check other C++ code repos?

Reply:
Got it, checked in other SONIC repo. Private member use "m_" as prefix. Public member/methods using Camel-Case naming. Will apply in latest diff.